src/components/DecisioningEngine/createSubscribeRulesetItems.js

```diff
@@ -28,41 +28,44 @@ const validateOptions = ({ options }) => {
 
 export default () => {
   let subscriptionHandler;
-  let surfaceIdentifier;
+  let surfacesFilter;
   let schemasFilter;
 
-  const run = ({ surface, schemas, callback }) => {
+  const run = ({ surfaces, schemas, callback }) => {
     subscriptionHandler = callback;
-    surfaceIdentifier = surface;
+    surfacesFilter = surfaces instanceof Array ? surfaces : undefined;
     schemasFilter = schemas instanceof Array ? schemas : undefined;
   };
 
   const optionsValidator = options => validateOptions({ options });
 
   const refresh = propositions => {
-    if (!subscriptionHandler || !surfaceIdentifier) {
+    if (!subscriptionHandler) {
       return;
     }
 
-    const result = propositions
-      .filter(payload => payload.scope === surfaceIdentifier)
-      .reduce((allItems, payload) => {
-        const { items = [] } = payload;
+    const result = {
+      propositions: propositions
+        .filter(payload =>
+          surfacesFilter ? surfacesFilter.includes(payload.scope) : true
+        )
+        .map(payload => {
+          const { items = [] } = payload;
+          return {
+            ...payload,
+            items: items.filter(item =>
+              schemasFilter ? schemasFilter.includes(item.schema) : true
```

Comment:
Shall we add check for invalid item ? 
`
if (!item || !item.schema) {
            return false; 
          }`

Reply:
Similar validation for payload above
```
 if (!payload || !payload.scope) {
        return false; 
      }
```

Reply:
@shandilya3 not clear on what you are suggesting here and what it accomplishes.  Can you share more?

Reply:
item will never be undefined, nor will payload.

Reply:
In that case the check is not needed.